sw/device/lib/dif/README.md

```diff
@@ -74,26 +93,121 @@ there are some relaxations of these rules for them described at the end.
+
+  Additional, specific return code constants must all be defined after
+  these three general constants, and may cover more specific forms of the
+  return codes defined above, including more specific reasons arguments are
+  invalid.
+* `dif_<ip>_toggle_t` is an enum representing an enabled or disabled state. This
+  must define exactly two constants, in no particular order:
+  * `kDif<ip>ToggleEnable`, indicating an enabled state.
+  * `kDif<ip>ToggleDisable`, indicating an enabled state.
+
+  This type is intended to be used as a better-named replacement for `bool`,
+  for operations which are setting wither a behavior is enabled or disabled
+  (such as whether an interrupt is maskable). Values of this type should *not* be
+  used as `if`-statement conditions.
+
+  If no function requires this type, it may be omitted.
+
+#### Lifecycle Functions
+The following functions form the basic functionality for initializing and
+handling the lifetime of a handle.
+
+* `result_t dif_<ip>_init(dif_<ip>_params_t params, dif_<ip>_t *handle);`
+  is a function that initializes `handle` in an implementation-defined way, but
+  does not perform any hardware operations. `handle` may point to uninitialized
+  data on function entry, but if the function returns `Ok`, then `handle` must
+  point to initialized data.
+* `result_t dif_<ip>_configure(const dif_<ip>_t *handle, dif_<ip>_config_t
+  config);` is a function that configures the hardware managed by `handle` with
+  runtime parameters in an implementation-defined way. This function should be
+  "one-off": it should only need to be called once for the lifetime of the handle.
+
+  If there is no meaningful state to configure, this function may be omitted.
+
+#### Interrupt Management
+
+The following types and functions are a standard inteface for peripherals that
+provide a collection of `INTR_ENABLE`, `INTR_STATE`, and `INTR_TEST` registers
+for interrupt management. A DIF library for a peripheral providing such
+registers must provide this interface.
+
+If a peripheral is defined with `no_auto_intr_regs: true`, this exact API is not
+required even of the above registers are provided (though DIF libraries are
+encouraged to follow it where it makes sense).
+
+* `dif_<ip>_irq_t` is an enum that lists all of the interrupt types for this
+  peripheral. A DIF library may opt to use another pre-existing type instead,
+  when, for example, interrupt types are coupled to a distinct
+  peripheral-specific concept. In that case, all occurences of `dif_<ip>_irq_t`
+  belkow should be replaced with this typoe.
+* `result_t dif_<ip>_irq_is_pending(const dif_<ip>_t *handle, dif_<ip>_irq_t
+  irq_type, bool *is_pending);` checks whether a specific interrupt is
+  pending (i.e., if the interrupt has been asserted but not yet cleared).
+* `result_t dif_<ip>_irq_clear(const dif_<ip>_t *handle, dif_<ip>_irq_t
+  irq_type);` clears a specific interrupt (and does nothing if the interrupt is
+  not currrently pending).
```

Comment:
I think this should be `dif_<ip>_irq_acknowledge(…)` as you're acknowledging you've processed the interrupt - `clear` has lots of meanings which can be confused (you might be clearing the bit that enables that interrupt in INTR_ENABLE)

Reply:
That's a good idea actually. I personally prefer `ack()`, since it's a pretty common abbreviation and a lot of these names are a little long. What do you think?

Reply:
I think `acknowledge`, it's not too many extra letters, you'll cope :)